# SPDX-FileCopyrightText: 2019-2020 Smart Robotics Lab, Imperial College London
# SPDX-FileCopyrightText: 2019 Anna Dai
# SPDX-FileCopyrightText: 2020 Sotiris Papatheodorou
# SPDX-License-Identifier: CC0-1.0

# Supereight ROS node settings
supereight_ros:
  enable_tracking: true
  enable_rendering: true
  enable_rgb: false
  input_res: [640, 480]
  pose_topic_type: "geometry_msgs::PoseStamped"
  pose_buffer_size: 600
  depth_buffer_size: 60
  rgb_buffer_size: 60
  max_timestamp_diff: 0.02

# Supereight settings
supereight:
  general:
    sequence_name:              ""
    sequence_type:              "ros"
    output_render_path:         ""
    enable_meshing:             false
    output_mesh_path:           ""
    # Rates
    tracking_rate:              1
    integration_rate:           1
    rendering_rate:             4
    meshing_rate:               0
    fps:                        0.0
    # Other
    drop_frames:                false
    max_frame:                  -1
    icp_threshold:              1e-5
    bilateral_filter:           false
    pyramid:                    [10, 5, 4]

  map:
    size:                       256
    dim:                        10.24
    t_MW_factor:                [0.501, 0.501, 0.501]

  sensor:
    intrinsics:                 [205.46963709898583, 205.46963709898583, 319.5, 239.5]
    downsampling_factor:        1
    T_BC:                       [1, 0, 0, 0,
                                 0, 1, 0, 0,
                                 0, 0, 1, 0,
                                 0, 0, 0, 1]
    init_T_WB:                  [1, 0, 0, 0,
                                 0, 1, 0, 0,
                                 0, 0, 1, 0,
                                 0, 0, 0, 1]
    near_plane:                 0.4
    far_plane:                  4.0

